actors/market/src/lib.rs

```diff
@@ -234,16 +235,34 @@ impl Actor {
             ));
         }
 
+        struct CheckedDeal {
+            proposal: DealProposal,
+            check_passed: bool,
+        }
+        // Deals that passed `AuthenticateMessage` and other state-less checks.
+        let mut checked_deals: Vec<CheckedDeal> = Vec::with_capacity(params.deals.len());
+
         let baseline_power = request_current_baseline_power(rt)?;
         let (network_raw_power, _) = request_current_network_power(rt)?;
 
+        for (di, deal) in params.deals.into_iter().enumerate() {
```

Comment:
Please add a comment here that these checks need to happen before loading state, due to the possibility of recursion from authentication.